front-end/src/App.vue

```diff
@@ -0,0 +1,120 @@
+<script setup>
+import EndpointDropdown from './components/EndpointDropdown.vue';
+import Toast from 'primevue/toast';
+import { computed, watch, onMounted } from 'vue';
+import { RouterLink, RouterView } from 'vue-router';
+import { useAPIGatewayStore } from './stores/apiGateway';
+import { useToast } from 'primevue/usetoast';
+
+const apiGateway = useAPIGatewayStore();
+const toast = useToast();
+
+const apiRoutes = computed(() => {
+  return apiGateway.routes;
+});
+
+const currentAPIRoute = computed(() => {
+  return apiGateway.currentRoute;
+});
+
+const responseStatus = computed(() => {
+  return apiGateway.recentResponseStatus;
+});
+
+const updateAPIRoute = (newRoute) => {
+  apiGateway.setRoute(newRoute);
+  apiGateway.getData();
+}
+
+watch(responseStatus, (newStatus) => {
+  const catagory = parseInt(newStatus.status.toString().charAt(0), 10);
+  if (catagory === 2) {
+    toast.add({ severity: 'success', summary: 'Success!', detail: `${newStatus.method} request for endpoint: ${newStatus.route} succeeded`, life: 3000 });
+  } else if (catagory == 4 || catagory == 5) {
+    toast.add({ severity: 'error', summary: 'Failure!', detail: `${newStatus.method} request for endpoint: ${newStatus.route} failed`, life: 3000 });
+  }
+});
+
+onMounted(() => {
+  const ROUTE_STORAGE_KEY = 'API#ROUTE'
+  const storedRoute = JSON.parse(localStorage.getItem(ROUTE_STORAGE_KEY));
+  if(storedRoute == null) {
+    apiGateway.getData();
+  } else {
+    apiGateway.setRoute(storedRoute);
+  }
+  
+});
+</script>
+
+<template>
+  <div class='app-wrapper'>
+    <Toast />
+    <div class='header-wrapper'>
+      <header class='navigation-wrapper'>
+        <h1 class='title'>Joshua Daigle Assessment</h1>
+        <nav class='top-navigation'>
+          <RouterLink class='router-link' to='/'>View</RouterLink>
+          <RouterLink class='router-link' to='/add'>Add</RouterLink>
+          <EndpointDropdown @route-changed='updateAPIRoute' :routes='apiRoutes' :current-route='currentAPIRoute'></EndpointDropdown>
+        </nav>
+      </header>
+    </div>
+    <div class='view-wrapper'>
+      <RouterView />
+    </div>
+  </div>
+</template>
+
+<style scoped>
```

Comment:
The way I understand scoped is that none of these styles will 'leak' into child components.

You are setting a font family for app wrapper, does that mean child components will have a different font family/

Question, how do you make the decision to put css in the scoped context vs in the global stylesheet

Reply:
I'm actually not sure why I have that there, font family should definitely be in the global stylesheet. 

A component using scoped style will have it's style "leak" into only its direct children.

As a rule of thumb, I am careful about placing css in the global scope. This is to avoid future headaches from things like naming collisions and inheritance conflicts. Only css which I want to be the "default behaviour" should go there. In this case, font family is a perfect example. 

Overall, the encapsulation provided by scoped styling is one of my favourite features of Vue. It makes maintaining css as the project grows in size much easier to work with.